src/RevPrefetcher.cc

```diff
@@ -102,6 +108,13 @@ bool RevPrefetcher::InstFetch(uint64_t Addr, bool &Fetched, uint32_t &Inst){
 }
 
 void RevPrefetcher::Fill(uint64_t Addr){
+
+  // determine if the address is 32bit aligned
+  if((Addr & 0x3ull)){
+    // not 32bit aligned, adjust the base address by 2 bytes
+    Fill(Addr-2);
```

Comment:
do we want to subtract or mask to do the alignment?

Reply:
subtracting aligns it to the next lowest aligned word.  is that sufficient?

Reply:
So if Addr == 0x1234567F, subtracting 2 gives 0x12234567D which is still not aligned, right?

Reply:
that's assuming that the incoming address request is not aligned.  in theory, it should be, but I see your point.  If we masked off the lower two bits, it would be aligned.  0x123456F would be 0x1234567C

Reply:
see the new code in [81d19f9](https://github.com/tactcomplabs/rev/pull/31/commits/81d19f92ed0f6342fc43d3a70cf6538c3a5bbccb)

Reply:
Looks good!